src/giftCards/GiftCardUpdate/GiftCardHistory/hooks/useGiftCardHistoryEvents.test.tsx

```diff
@@ -0,0 +1,105 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { renderHook } from "@testing-library/react-hooks";
+import React from "react";
+
+import { giftCardsMocks } from "../../../../../testUtils/mocks/giftCards";
+import { useUser } from "../../../../auth";
+import { useGiftCardEventsQuery } from "../../../../graphql";
+import { GiftCardDetailsContext } from "../../providers/GiftCardDetailsProvider";
+import useGiftCardHistoryEvents from "./useGiftCardHistoryEvents";
+
+// Mock the dependencies
+jest.mock("@dashboard/auth");
+jest.mock("@dashboard/graphql");
+
+const mockUseUser = useUser as jest.Mock;
+const mockUseGiftCardEventsQuery = useGiftCardEventsQuery as jest.Mock;
+
+const mockGiftCard = {
+  ...giftCardsMocks[0],
+  id: "giftCardId",
+};
+const mockUser = {
+  userPermissions: [{ code: "MANAGE_USERS" }, { code: "MANAGE_APPS" }],
+};
+
+mockUseUser.mockReturnValue({ user: mockUser });
+mockUseGiftCardEventsQuery.mockImplementation(value => {
+  if (value.skip) {
+    return {
+      data: undefined,
+    };
+  }
+
+  if (!value.variables.canSeeApp || !value.variables.canSeeUser) {
+    return {
+      data: undefined,
+    };
+  }
+
+  return {
+    data: {
+      giftCard: {
+        events: [{ id: "event1" }, { id: "event2" }],
+      },
+    },
+  };
```

Comment:
this code actually should be tested as well - you provide here **your own implementation** for the hook that comes from library (read the comments below regarding using wrappers)